backend/src/announcement/announcement.service.ts

```diff
@@ -36,42 +41,77 @@ export class AnnouncementServices {
     }
   }
   public async createAnnouncement(
+    user: User,
     data: AnnouncementDTO
   ): Promise<AnnouncementDTO> {
     try {
       const announcement = await this.prisma.announcement.create({
         data,
       });
+      this.eventEmitter.emit('create.logs', {
+        entityName: 'announcement',
+        action: ActionENUM.ADDED,
+        username: user.name,
+        priority: PriorityENUM.IMPORTANT,
+      });
+
       return AnnouncementServices.convertToDTO(announcement);
     } catch (error) {
       throw new BadRequestException();
     }
   }
   public async updateAnnouncement(
+    user: User,
     id: string,
     data: AnnouncementDTO
   ): Promise<AnnouncementDTO> {
     try {
+      // DONT KNOW IF THIS IS THE BEST WAY TO DO IT
+      const oldValue = await this.prisma.announcement.findUnique({
+        where: {
+          id,
+        },
+      });
       const announcement = await this.prisma.announcement.update({
         where: {
           id,
         },
         data,
       });
+
+      this.eventEmitter.emit('create.logs', {
+        entityName: 'announcement',
+        action: ActionENUM.EDITED,
+        username: user.name,
+        priority: PriorityENUM.IMPORTANT,
+        oldValue: JSON.stringify(oldValue),
```

Comment:
used `JSON.stringify` and `JSON.parse` on fields `oldValue` and `newValue` since I'm taking the entire JSON response from the database.